diff --git a/resources/niri-session b/resources/niri-session
index e0c71ffeaf..49464d4f86 100755
--- a/resources/niri-session
+++ b/resources/niri-session
@@ -33,7 +33,7 @@ if hash systemctl >/dev/null 2>&1; then
     systemctl --user reset-failed
 
     # Import the login manager environment.
-    systemctl --user import-environment
+    systemctl --user import-environment $(printenv | cut -d'=' -f1 | tr '\n' ' ')
 
     # DBus activation environment is independent from systemd. While most of
     # dbus-activated services are already using `SystemdService` directive, some
